src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/commands.py

```diff
@@ -0,0 +1,107 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands import CliCommandType
+from ._client_factory import (get_mediaservices_client, get_transforms_client,
+                              get_assets_client, get_jobs_client, get_streaming_locators_client,
+                              get_streaming_policies_client)
+from ._exception_handler import (build_exception_wrapper)
+
+
+def load_command_table(self, _):  # pylint: disable=too-many-locals, too-many-statements
+    def get_sdk(operation, client_factory):
+        return CliCommandType(
+            operations_tmpl='azure.mediav3.operations#{}Operations.'.format(operation) + '{}',
+            client_factory=client_factory,
+            exception_handler=build_exception_wrapper()
+        )
+
+    def get_custom_sdk(custom_module, client_factory):
+        return CliCommandType(
+            operations_tmpl='azure.cli.command_modules.ams.operations.{}#'.format(custom_module) + '{}',
+            client_factory=client_factory,
+            exception_handler=build_exception_wrapper()
+        )
```

Comment:
The command registration seems much more verbose than I usually see, and I think the culprit may be this pattern. Did you emulate this pattern from an existing module?

Reply:
Yes, we emulate this pattern from the `storage` module:

https://github.com/Azure/azure-cli/blob/105d22c0905dc50a08251e7e0530167a4935583a/src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/commands.py#L32

Should we consider changing this implementation?

Reply:
You really should not follow storage.  Nearly any other module would be much simpler.  Please look at network and see if that pattern is simpler. It should be.